invenio_records_resources/services/files/components/metadata.py

```diff
@@ -25,7 +26,18 @@ def init_files(self, identity, id, record, data):
         validated_data = schema.load(data)
         for file_metadata in validated_data:
             temporary_obj = deepcopy(file_metadata)
-            record.files.create(temporary_obj.pop("key"), data=temporary_obj)
+            file_type = temporary_obj.pop("storage_class", None)
+            transfer = Transfer.get_transfer(file_type)
+            file_record = transfer.init_file(record, temporary_obj)
+            if Transfer.fetch_file(file_record.file):
```

Comment:
Small note about wording. 

When I first read this, `fetch_file` looked to me like a function that would fetch the file. However, after reading the implentation, I understand that it's a validation on whether the file is fetchable from an external source (without validating specifically for file type here).

Perhaps something like `is_____()` would be easier to read?

Reply:
Agree, or something like `needs_...`.

Reply:
I have move it inside the `FetchTransfer` object. Passing the `uow` is still not optimal. My gut feeling tells me that we will need it for other transfer types, and most important that way we do not expose the specific behaviour outside of the Transfer type.